src/client/connection.rs

```diff
@@ -108,12 +108,15 @@ impl HttpError {
                 } else if e.is_timeout() {
                     kind = HttpErrorKind::Timeout;
                 }
-                break;
```

Comment:
I think this means that now we will potentially recategorize an error already categorized by hyper, this seems undesirable.

Reply:
I think the problematic part is that we break regardless whether any of the conditions (`is_closed`, `is_timeout`...) are true. It would probably be fine to only break if one of them is satisfied.